editoast/src/views/infra/edition.rs

```diff
@@ -54,6 +80,753 @@ pub async fn edit<'a>(
+    let right_tracksection = TrackSection {
+        id: Identifier::from(right_tracksection_id),
+        length: tracksection.length - distance,
+        geo: tracksection_data.right_geo.as_ref().clone(),
+        sch: tracksection_data.right_geo.as_ref().clone(),
+        slopes: tracksection
+            .slopes
+            .iter()
+            .filter(|e| e.end >= distance)
+            .map(|e| {
+                let mut item = e.clone();
+                if item.begin < distance {
+                    item.begin = distance;
+                } else {
+                    item.begin = item.begin - distance;
+                }
+                item.end = item.end - distance;
+                return item;
+            })
+            .collect_vec(),
+        curves: tracksection
+            .curves
+            .iter()
+            .filter(|e| e.end >= distance)
+            .map(|e| {
+                let mut item = e.clone();
+                if item.begin < distance {
+                    item.begin = distance;
+                } else {
+                    item.begin = item.begin - distance;
+                }
+                item.end = item.end - distance;
+                return item;
+            })
+            .collect_vec(),
+        loading_gauge_limits: tracksection
+            .loading_gauge_limits
+            .iter()
+            .filter(|e| e.end >= distance)
+            .map(|e| {
+                let mut item = e.clone();
+                if item.begin < distance {
+                    item.begin = distance;
+                } else {
+                    item.begin = item.begin - distance;
+                }
+                item.end = item.end - distance;
+                return item;
+            })
+            .collect_vec(),
+        ..tracksection.clone()
+    };
+    // track link
+    let mut ports = HashMap::new();
+    ports.insert(
+        "A".into(),
+        TrackEndpoint {
+            track: Identifier::from(left_tracksection_id),
+            endpoint: Endpoint::End,
+        },
+    );
+    ports.insert(
+        "B".into(),
+        TrackEndpoint {
+            track: Identifier::from(right_tracksection_id),
+            endpoint: Endpoint::End,
+        },
+    );
+    let track_link = Switch {
+        id: Identifier::from(Uuid::new_v4()),
+        switch_type: Identifier::from("link"),
+        group_change_delay: 0.0,
+        ports,
+        ..Switch::default()
+    };
+
+    // Compute operations
+    // ~~~~~~~~~~~~~~~~~~~~~~~
+    // Firstly, we create the two newly tracks
+    let mut operations: Vec<Operation> = [
+        Operation::Create(Box::new(RailjsonObject::TrackSection {
+            railjson: left_tracksection.into(),
+        })),
+        Operation::Create(Box::new(RailjsonObject::TrackSection {
+            railjson: right_tracksection.into(),
+        })),
+        Operation::Create(Box::new(RailjsonObject::Switch {
+            railjson: track_link.into(),
+        })),
```

Comment:
These three operations could be calculated in a separate function.

Reply:
agree, but it's only used here and simple to understand in the flow of the function.
So IMHO, it's not needed.